### DISCUSSION TITLE
AttributeError: module 'torch' has no attribute 'pi'

### DISCUSSION THREAD

**Discussion by MussaSman** [ourownstory/neural_prophet#584] [Q&A - get help using NeuralProphet] [answered]
Hello,
I am trying to fit model. I get the AttributeError: module 'torch' has no attribute 'pi' error when fit model.

df_nn_tail = df_nn.tail(21600)
df_nn_tail.head()

2021-10-08 06:11:05 | 3.27645
2021-10-08 06:12:05 | 3.34651
2021-10-08 06:13:05 | 3.63989
2021-10-08 06:14:05 | 3.64379
2021-10-08 06:15:05 | 3.59195

m = NeuralProphet()
metrics = m.fit(df_nn_tail, freq="10min")

INFO - (NP.df_utils._infer_frequency) - Major frequency T corresponds to 94.769% of the data.
WARNING - (NP.df_utils._infer_frequency) - Defined frequency 10min is different than major frequency T
INFO - (NP.config.init_data_params) - Setting normalization to global as only one dataframe provided for training.
INFO - (NP.utils.set_auto_seasonalities) - Disabling yearly seasonality. Run NeuralProphet with yearly_seasonality=True to override this.
INFO - (NP.config.set_auto_batch_epoch) - Auto-set batch_size to 64
INFO - (NP.config.set_auto_batch_epoch) - Auto-set epochs to 46
 91%|█████████▏| 289/316 [00:01<00:00, 195.43it/s]
INFO - (NP.utils_torch.lr_range_test) - lr-range-test results: steep: 3.98E-02, min: 6.74E-01
 93%|█████████▎| 295/316 [00:01<00:00, 187.96it/s]
INFO - (NP.utils_torch.lr_range_test) - lr-range-test results: steep: 3.49E-02, min: 7.69E-01
 92%|█████████▏| 290/316 [00:01<00:00, 192.06it/s]
INFO - (NP.utils_torch.lr_range_test) - lr-range-test results: steep: 3.49E-02, min: 9.36E-01
INFO - (NP.forecaster._init_train_loader) - lr-range-test selected learning rate: 3.65E-02
  0%|          | 0/46 [00:00<?, ?it/s]

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_34152/3781958549.py in <module>
      1 m = NeuralProphet()
----> 2 metrics = m.fit(df_nn_tail, freq="10min")

~\AppData\Local\Programs\Python\Python39\lib\site-packages\neuralprophet\forecaster.py in fit(self, df, freq, validation_df, progress, minimal)
    458                 metrics_df = None
    459             else:
--> 460                 metrics_df = self._train(df_dict, progress=progress)
    461         else:
    462             df_val_dict, _ = df_utils.prep_copy_df_dict(validation_df)

~\AppData\Local\Programs\Python\Python39\lib\site-packages\neuralprophet\forecaster.py in _train(self, df_dict, df_val_dict, progress)
   1472                 val_metrics.reset()
   1473             # run epoch
-> 1474             epoch_metrics = self._train_epoch(e, loader)
   1475             # collect metrics
   1476             if validate:

~\AppData\Local\Programs\Python\Python39\lib\site-packages\neuralprophet\forecaster.py in _train_epoch(self, e, loader)
   1285             loss = self.config_train.loss_func(predicted, targets)
   1286             # Weigh newer samples more.
-> 1287             loss = loss * self._get_time_based_sample_weight(t=inputs["time"])
   1288             loss = loss.mean()
   1289             # Regularize.

~\AppData\Local\Programs\Python\Python39\lib\site-packages\neuralprophet\forecaster.py in _get_time_based_sample_weight(self, t)
   1264             time = torch.maximum(torch.zeros_like(time), time)
   1265             time = torch.minimum(torch.ones_like(time), time)  # time = 0 to 1
-> 1266             time = torch.pi * (time - 1.0)  # time =  -pi to 0
   1267             time = 0.5 * torch.cos(time) + 0.5  # time =  0 to 1
   1268             # scales end to be end weight times bigger than start weight

AttributeError: module 'torch' has no attribute 'pi'

**Comment by jSwords91** [comment]
Hi. I too had this issue. We can get around it by using:
`import math

  import torch

  torch.pi = math.pi `

Not ideal, but will get you by for now until the problem is resolved.

**Reply by ourownstory** [reply]
Thank you for posting this workaround!

**Comment by ourownstory** [comment, marked as answer]
The issue has been addressed in 0.3.2
